### DISCUSSION TITLE
[ 2021년 06월 07일 (월) ] 피어세션 내용

### DISCUSSION THREAD

**Discussion by sglee487** [bcaitech1/p4-mod-model_diet#32] [General]
* squeezenet이 alexnet에서 파라미터 수만 줄인거라 거의 학습이 안됨.. dropout 같은거 설정 했는데도 잘 안되었다. f1 0.2에서 왔다갔다 거린 듯.
* 최소 45epoch이어야 f1 넘었다. 1epoch당 2분 30초 정도.
* hyperparameter 수정한것도 있음. 한 주기를 40~50 에폭으로 함. local minimum을 찾으려면 이렇게 해야 함.
* augmentation. simple_augment_train 에서 나머지 다 해봤는데 점수가 잘나옴. resize에서 1.2 빼고 randomcrop을 아얘 뺐다. horizen flip, vertical flip, to tensor, normalize 로 했더니 f1 점수 0.1가량 오름.
* 이미지 사이즈 (3, 244, 244) -> (3, 64, 64) 로 했더니 MACs가 41810537 -> 3421577로 10배가량 줄었고 점수는 0.7~0.8 정도 나왔음. 하지만 그만큼 훈련을 더 많이 해야 하는 듯 하다. 244 사이즈로는 15epoch만 해도 됐던게 40epoch을 했었다.
* cosine restart로 하니 100epoch이어도 올라가긴 하더라.

## 주말(지금까지) 한 것들
* 상건: rank 개념, wandb 실험한거 (아직 결론을 못냈음)(https://wandb.ai/sglee487/pstage4?workspace=user-sglee487)
* 연세님: 자소서, 지원서 제출. 오늘 강의내용 이해 한된거 있었음
* 석진님: TPE 알고리즘 알아봄. 또 알고리즘 문제 풀어봄. special mission 보고있음. 너무 복잡해서 baseline code에서 늘리는 방식으로 정리, 저장하고 하고있는 중.
* 황훈님: squeezenet 학습해서 돌리고, 포트폴리오 만듬. squeezenet은 별로 좋진 않은 듯.
* 용우님: MACs. mobilenetv3 small 에서 임의로 줄여서 이제 경량화만 하면 될듯. multiple 1.0으로 고정해서 사용. stride도 기존 mobilenetv3 large, small을 보고 참고하면서 하고 있다.
* 웅희님: batch 할 때마다 꺼내서 결과 보고 다시 올리고 하는걸텐데 의미가 있을까? 참조형식으로 해보려고 했는데도 잘 안됨. invertedresidual v3랑 dwconv 말고 다른것도 같이 사용했더니 MACs가 좀 들어난듯. adam이 잘 안되는 것 같다.

## 하고싶은 것
* 상건: 모델 개조한다면 shufflenetv2 0.5 개조해서 할것 같다. 근데 할게 많아서 언제 할 수 있을지.. 포트폴리오 쓴다.
* 연세님: 기업지원을 하겠다.
* 석진님: 포트폴리오 쓸려고 한다.
* 황훈님: optuna에 도전해보겠다.
* 웅희님: image 64로 하고 pretrained weight로 불러서 해보겠다. train 돌려보고..
* 용우님: 포트폴리오 하려고 한다.
